accepted/00101-health-check-tool.md

```diff
@@ -0,0 +1,282 @@
+- Integration with Monitoring Stack. Enhance monitoring formula to enable Health Check monitoring.
+- Reuse configuration from "standalone" version.
+
+#### Saline
+- Provide "Saline" inside a separated container that attach to Salt Master TCP sockets from "uyuni-server" container.
+- Alternatively, explore sharing Salt Master IPC sockets between containers.
+
+
+# How is the exporter going to access to uyuni-server container? For example, to query the database.
+As mentioned, the exporter needs to have access to Uyuni server container. It does need to query the DB and also execute some Salt runners jobs to gather metrics. If the "exporter" runs in a separated container than the Salt Master, then we can make the "exporter" container to run in the same network than the "uyuni-server", then TCP sockets are available to use.
+
+There are some Salt runners jobs that requires access to Salt Master Event publisher, currently exposed via IPC socket. This makes it tricky access these sockets from a different container. Some metrics might not be available until we move to "TCP" sockets for the Salt Master Event publisher in this case.
+
+Alternatively, we could make the "exporter" as part of the server-image, even if not running by default. Then it is up to the Healh Check Tool or Monitoring formula to start it. This would solve the current issue of not having access to the IPC sockets.
+
+
+# Security considerations
+In the current design, we are exposing metrics via Promotheus and making them available in Grafana, and more importantly we are exposing logs messages via Loki/LogCLI to CLI and Grafana users. It is important to notice that after running this tool, and until related containers are destroyed, the Grafana Dashboards (and other components like Prometheus and Loki/LogCLI) are exposing metrics and logs messages that may contain sensitive data and information to any non-root user in the system or to anyone that have access to this host in the network.
+
+Promtail pipeline definition can be enhanced to use [replace](https://grafana.com/docs/loki/latest/send-data/promtail/stages/replace/) function, to hidden sensitive data from logs.
+
+Authentication for backend for Prometheus and Loki is usually delegated to an authentication reverse proxy.
+
+
+# Drawbacks
+
+[drawbacks]: #drawbacks
+
+
+* Resource Usage: Running additional services for log aggregation, analysis, and enhanced alerting could increase the resource demands on the server, including CPU, memory, and storage. This might necessitate upgrades to existing hardware or reevaluation of resource allocation in cloud environments to ensure optimal performance.
+
+* Learning Curve: System administrators and developers may need to familiarize themselves with Loki/Promtail's log querying language (LogQL) and Grafana's alerting mechanisms. The learning curve associated with these tools could slow down initial adoption and efficiency gains.
```

Comment:
Is the learning curve for Loki shallower than for other (more popular) solutions? If it's as steep as the competition, I don't see the learning curve as a drawback.

That said, what are the plans for the documentation to help users adopting the new monitoring capabilities?

Reply:
> Is the learning curve for Loki shallower than for other (more popular) solutions? If it's as steep as the competition, I don't see the learning curve as a drawback.
> 
> That said, what are the plans for the documentation to help users adopting the new monitoring capabilities?

Yeah, I see your point - other popular solutions would have also similar learning curve, so in this sense, learning curve is also a drawback if we go with different solution.

Actually, as Grafana is already part of our Monitoring stack, that would probably help with the learning curve.

I'm moving the "learning curve" to a separated "Learning curve and Documentation" section outside of the "drawbacks".